src/docfx/Errors.cs

```diff
@@ -9,18 +9,27 @@ namespace Microsoft.Docs.Build
     internal static class Errors
     {
         public static DocfxException ConfigNotFound(string docsetPath)
-            => new DocfxException("config-not-found", $"Cannot find docfx.yml at '{docsetPath}'");
+            => new DocfxException(ReportLevel.Error, "config-not-found", $"Cannot find docfx.yml at '{docsetPath}'");
 
         public static DocfxException InvalidConfig(string configPath, Exception e)
-            => new DocfxException("invalid-config", $"Error parsing docset config: {e.Message}", configPath, innerException: e);
+            => new DocfxException(ReportLevel.Error, "invalid-config", $"Error parsing docset config: {e.Message}", configPath, innerException: e);
 
         public static DocfxException CircularReference<T>(T filePath, IEnumerable<T> dependencyChain)
-            => new DocfxException("circular-reference", $"Found circular reference: {string.Join(" --> ", dependencyChain)} --> {filePath}", filePath.ToString());
+            => new DocfxException(ReportLevel.Error, "circular-reference", $"Found circular reference: {string.Join(" --> ", dependencyChain)} --> {filePath}", filePath.ToString());
 
         public static DocfxException YamlHeaderNotObject(object filePath, bool isArray)
-            => new DocfxException("yaml-header-not-object", $"Expect yaml header to be an object, but got {(isArray ? "an array" : "a scalar")}", filePath.ToString());
+            => new DocfxException(ReportLevel.Warning, "yaml-header-not-object", $"Expect yaml header to be an object, but got {(isArray ? "an array" : "a scalar")}", filePath.ToString());
 
         public static DocfxException InvalidYamlHeader(Document file, Exception ex)
-            => new DocfxException("invalid-yaml-header", ex.Message, file.ToString());
+            => new DocfxException(ReportLevel.Warning, "invalid-yaml-header", ex.Message, file.ToString());
+
+        public static DocfxException LinkIsEmpty(Document file)
+            => new DocfxException(ReportLevel.Warning, "link-is-empty", "Link is empty", file.ToString());
+
+        public static DocfxException LinkIsAbsolute(Document file, string link)
```

Comment:
>LinkIsAbsolute [](start = 37, length = 14)

why absolute link need report exception?

Reply:
it is link to absolute file path like: ‘c:/a.md’.

---
In reply to: [193958862](https://github.com/dotnet/docfx/pull/2857#discussion_r193958862) [](ancestors = 193958862)